keepalived-vip/utils.go

```diff
@@ -239,3 +271,81 @@ func parseNsName(input string) (string, string, error) {
 
 	return nsName[0], nsName[1], nil
 }
+
+func parseNsSvcLVS(input string) (string, string, string, error) {
+	nsSvcLb := nsSvcLbRegex.FindStringSubmatch(input)
+	if len(nsSvcLb) != 6 {
+		return "", "", "", fmt.Errorf("invalid format (namespace/service name[:NAT|DR]) found in '%v'", input)
+	}
+
+	ns := nsSvcLb[1]
+	svc := nsSvcLb[2]
+	kind := nsSvcLb[3]
+
+	if ns == "" {
+		ns = nsSvcLb[4]
+	}
+
+	if svc == "" {
+		svc = nsSvcLb[5]
+	}
+
+	if kind == "" {
+		kind = "NAT"
+	}
+
+	if !lvsRegex.MatchString(kind) {
+		return "", "", "", fmt.Errorf("invalid LVS method. Only NAT and DR are supported: %v", kind)
+	}
+
+	return ns, svc, kind, nil
+}
+
+type nodeSelector map[string]string
+
+func (ns nodeSelector) String() string {
+	kv := []string{}
+	for key, val := range ns {
+		kv = append(kv, fmt.Sprintf("%v=%v", key, val))
+	}
+
+	return strings.Join(kv, ",")
+}
+
+func parseNodeSelector(data map[string]string) string {
+	return nodeSelector(data).String()
+}
+
+func waitForPodRunning(kubeClient *unversioned.Client, ns, podName string, interval, timeout time.Duration) error {
+	condition := func(pod *api.Pod) (bool, error) {
+		if pod.Status.Phase == api.PodRunning {
+			return true, nil
+		}
+		return false, nil
+	}
+
+	return waitForPodCondition(kubeClient, ns, podName, condition, interval, timeout)
+}
+
+// waitForPodCondition waits for a pod in state defined by a condition (func)
+func waitForPodCondition(kubeClient *unversioned.Client, ns, podName string, condition func(pod *api.Pod) (bool, error),
+	interval, timeout time.Duration) error {
+	return wait.PollImmediate(interval, timeout, func() (bool, error) {
```

Comment:
this error is useless, suggest bubbling it up as something clearer. IT'll basically say: timed out waiting for condition, suggest something like: timed out waiting to observe own status as Running.